apps/site/assets/ts/schedule/components/schedule-finder/ScheduleTable.tsx

```diff
@@ -1,12 +1,36 @@
 import React, { ReactElement } from "react";
-import { ServiceScheduleInfo } from "../__schedule";
+import { ServiceScheduleInfo, RoutePatternWithShape } from "../__schedule";
 import TableRow from "./TableRow";
 
-const ScheduleTable = ({
-  schedule
-}: {
+interface Props {
   schedule: ServiceScheduleInfo;
-}): ReactElement<HTMLElement> => {
+  routePatterns: RoutePatternWithShape[];
+}
+
+const isSchoolTrip = (
+  routePatternsById: {
+    [key: string]: RoutePatternWithShape;
+  },
+  routePatternId: string
+): boolean =>
+  (
+    (routePatternsById[routePatternId] &&
+      routePatternsById[routePatternId].time_desc) ||
+    ""
+  ).match(/school/gi) !== null;
+
+const ScheduleTable = ({
+  schedule,
+  routePatterns
+}: Props): ReactElement<HTMLElement> => {
+  const routePatternsById = routePatterns.reduce(
+    (accumulator, routePattern) =>
+      Object.assign({}, accumulator, { [routePattern.id]: routePattern }),
```

Comment:
and the spread operator 😉 
(how much am I allowed to sass you on this PR and still have you know I'm kidding? I'll stop at this one)

Reply:
oh man -- will update this -- I keep forgetting about the spread operator!